### ISSUE TITLE
Default channel not removed from DB when only channel is deleted

### ISSUE THREAD

**Issue by jeroenpraat** [redmatrix/hubzilla#263]
When a member deletes his/her only channel (automatically the default channel), the default channel for that account is not removed from the database. When that member creates a new channel, the default channel is **not** changed to that new channel. The result is that when that member re-logins he/she is redirected to a deleted channel and can't do anything anymore.

**Comment by redmatrix** [comment]
See include/Contact.php:334 

   // if this was the default channel, set another one as default
    if($a->account['account_default_channel'] == $channel_id) {
        $r = q("select channel_id from channel where channel_account_id = %d and channel_removed = 0 limit 1",
            intval($a->account['account_id']),
            intval(PAGE_REMOVED));
        if ($r) {
            $rr = q("update account set account_default_channel = %d where account_id = %d",
                intval($r[0]['channel_id']),
                intval($a->account['account_id']));
            logger("Default channel deleted, changing default to channel_id " . $r[0]['channel_id']);
        }
        else {
            $rr = q("update account set account_default_channel = 0 where account_id = %d",
                intval($r[0]['channel_id']),
                intval($a->account['account_id']));
        }
    }

This should be doing the right thing. The first thing of course would be to verify if this bit of code is or isn't doing the right thing, and if it isn't fix it. We start by finding an undeleted channel and setting the default to that. If there are no undeleted channels we set it to 0. That should allow it to be set to a new channel if a new channel is created.

**Comment by jeroenpraat** [comment]
I don't have good access now to my VPS, so I have to take a look to this code later today. 
I had two hub members (one 2 months ago, one yesterday) reporting problem logging in with this scenario. After I checked the account table I noticed that the default channel was still pointing to the deleted channel, although a new one was created. Changing that manually in the database fixed their problems.
